x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -20,20 +20,23 @@
 
 public final class LogicalVerifier extends PostOptimizationPhasePlanVerifier<LogicalPlan> {
 
-    public static final LogicalVerifier INSTANCE = new LogicalVerifier();
```

Comment:
When working on micro optimizations I remember seeing some cost associated with initializing list of rules in various optimizers. Here `LogicalVerifier` looks much simpler, but I wonder if we could reuse a singe `static` object for local verifier and one for general? This would avoid creating new instances for each query execution. I acknowledge here it is much cheaper than actual rules initialization else where.